### DISCUSSION TITLE
TorBox v.0.5.4 - work in progress

### DISCUSSION THREAD

**Discussion by radio24** [radio24/TorBox#276] [Archive]
**24 NOV 2024**: TorBox v.0.5.4 is merged into the master branch because it is to time consuming to take care for two branches. However, there are things, which are still open and will be implemented. Also, image files and the documentation on https://www.torbox.ch/ will be updated in the next weeks and months.

The most important new features are the following:
- [TorBox on a Cloud](https://www.torbox.ch/?page_id=3687) -> Installation of TorBox on a cloud server with the new installation option --on_a_cloud.
- [TorBox mini ](https://www.torbox.ch/?p=3607)-> TorBox on a [Raspberry Pi Zero 2 W](https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/) with an [USB-A adapter](https://www.tindie.com/products/8086net/solderless-zero-dongle-for-raspberry-pi-zero/), which as a RNDIS/Ethernet gadget can plugged into a USB port.

Here is the todo-list for the further development:
- [x] In general: Update internal Bridges 
- [x] In general: Is it possible to disable or even remove the NetworkManager and ModemManager? NetworkManager is responsible that DHCP doesn't work for usb0. Currently, we test it with TorBox mini and it seems to work. We have to check, if this is also an option for Debian and Ubuntu installations.
- [x] In general: Does the creation or removal of onion services in some cases lead to the insertion of unnecessary teaching lines in torrc?
- [ ] In general: Selective restore of domain exclusion.
- [x] In general: `avahi-daemon` and `avahi-daemon.socket` have to be disabled (with `systemctl`) because it could lead to network disruptions.
- [ ] tor compiling under Debian and Ubuntu testen (maybe `--disable-unittests` is necessary)
- [ ] TorBox mini: Migrate "Reset TorBox’s configuration files" in the [Update and Maintenance sub-menu](https://www.torbox.ch/?page_id=1410).
- [ ] In general: Move "Reset TorBox’s configuration files" in the [Danger Zone](https://www.torbox.ch/?page_id=3445).
- [ ] TorBox mini: Decision on what should be available in the Config sub-menu.
- [x] TorBox mini: Decision on building and giving out a separate image file.
- [x] TorBox mini: Creating a script to convert a standard TorBox installation to a TorBox mini. The idea is to have the following path to build the images: TorBox --> image creating --> executing the script --> TorBox mini --> image creating
- [X] In general: Add the migration script into the [Danger Zone](https://www.torbox.ch/?page_id=3445).
- [X] TorBox on a Cloud: Writing a build-manual for `torbox.ch` (in progress)

**Comment by radio24** [comment]
Interesting: The installed additional network driver are working with a Raspberry Pi 4 but not with a Raspberry Pi 5. 🤔
Ok, re-installing the network drivers fixes the problem.

**Comment by radio24** [comment]
I had problems rebuilding TorBox on a [Raspberry Pi Zero 2 W](https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/). The problem is that the `cmdline.txt` file can not just be replaced. It has the following option, which changes depending on the running kernel: `root=OARTUUID=nnnnn`.

That could be why working with [Raspberry Pi OS "Bookworm" lite 64-bit](https://www.raspberrypi.com/software/operating-systems/) was unsuccessful - I will recheck it later.

**Comment by gozillah** [comment]
I read your most recent reply on [link](https://github.com/radio24/TorBox/issues/264#issuecomment-2023812799) a bit late, sorry for the delay. I will have more time end of April / beginning of May. 

I noticed your April 2024 page about TorBox mini on torbox.ch, in particular the last section (thanks for the flowers!), and updated my post [link](https://github.com/radio24/TorBox/issues/264#issuecomment-1934294319) with a reference to the appropriate adapter and assembly instructions. 

You mentioned that ungracefully shutting down the Pi5 may cause SD card problems: The small, white power up/down button close to the LED is assumed to prevent this.

I installed TorBox054 from your Github sources on a uSD card with Pi OS Lite 64 Bookworm on my Pi5. Doing this headless it looped during installation of OpenCV. I then discovered your comment in run_install.sh and successfully repeated the install via Ethernet. The install took only 30 minutes on the Pi5, some 20% of the time required for TorBox053 mini on Zero 2W.

In the TorBox054 run_install.sh the TorBox default SSID and hostname are still TorBox053.

During installation of the additional network drivers some warnings flashed by:

![Warnings](https://github.com/radio24/TorBox/assets/156078441/68deb81e-0796-466d-af11-75913be707c6)

You mentioned the drivers not working initially. I haven't checked this yet as I am still using the Ethernet port for Internet access (thanks to the problem with OpenCV). In [link](https://github.com/radio24/TorBox/issues/264#issuecomment-1909426307) you mentioned having solved the OpenCV problem.

In my oppinion the additional network drivers are essential for those who are using the TorBox access point with Internet connection via wlan1 (e.g. at a public hotspot).

The font color issue at the end of the Tor restart ("100% Done ...") is now well fixed, but the instruction to quit with "q" will have quickly disappeared beyond the top of the screen (in particular on a Pi5 ;-) before the "100% Done ..." is being displayed:

![100% Done](https://github.com/radio24/TorBox/assets/156078441/207bdb0a-bb4a-4492-beff-c7c7301a0127)

The processing and screen updating is simply too fast for humans. Ideally the "q" instruction should appear after the "100% Done ..." (if programmatically possible).

I now have a number of (test-)Pis with various TorBox versions and I would appreciate a TorBox version number somewhere on the main menu, e.g. at its top like here:

![image](https://github.com/radio24/TorBox/assets/156078441/814cf396-6fd6-4ec4-b26e-afdd32edb905)

As for user friendliness, I think the current TorBox offers sort of an expert mode. A significant amount of prior reading and knowledge is necessary to adapt the parameters safely to the user's situation and intentions. User friendlyness is currently good for experts but not for non-experts. Looking at standard applications I would suggest the following raw ideas (just for consideration):
- Initial setup after installation split into an expert mode (as is) and a simplified one-click setup for non-experts (intentionally lacking finesse).
- Parameter management split into an expert mode (as is) and a non-expert mode with scenarios, i.e. the non-expert selects a usage scenario leading to programmatically selected parameters (even if this non-expert mode will not cover all the features available in TorBox). Section "Is the use of a TorBox (or tor in general) considered secure?" [link](https://www.torbox.ch/?page_id=112) was inspirational to me. The four scenarios there could be a starting point for brainstorming.
- Alternatively the non-expert could be guided through a series of questions which will then lead to a parameter set. The current user interface is implementing this idea to a very limited extent.

These are just my first ideas about improving user frienlyness. I am by no means certain if their implementation will be possible, make sense and show the expected results ...

**Comment by gozillah** [comment]
This text should be in red and "Tor" with leading capital letter:


![Picture](https://github.com/radio24/TorBox/assets/156078441/3801a5b1-a55f-4ffa-b9c6-78e8668bcb32)

So far still no problems with Torbox 0.5.4 on Pi5.

A collegue asked for TorBox mini with Internet over Ethernet instead of WiFi. The ubiquitous Waveshare Ethernet/USB-hat is occupiying usb0 for connecting the two boards. I am looking for alternatives., e.g. [link](https://www.aliexpress.com/item/1005006740442783.html)

**Comment by radio24** [comment]
SORRY FOR TEMPORARY STOPPING WORKING ON TORBOX
At the beginning of 2024, I planned to introduce a revised version of TorBox in the middle of the year. Unfortunately, that didn't materialize. It is due to the massive pile of work in my new job. This will delay my work on TorBox for at least six months. However, I hope to resume work on TorBox at the end of 2024/beginning of 2025.
Sorry!
